app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,24 @@
+class CommentsController < ApplicationController
+  layout 'boilerplate'
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    @comment = Comment.new(comment_params)
+    @comment.user = current_user
+    if @comment.save
+      flash[:success] = 'Comment added successfully!'
+      redirect_to user_post_path(id: @comment.post_id, user_id: @comment.user_id)
+    else
+      flash.now[:error] = 'Error: Comment could not be saved!'
+      render :new, locals: { comment: @comment }
+    end
+  end
```

Comment:
- [ ] I will encourage you to check the video I made [here](https://www.loom.com/share/6509a44b6fc5423eac04ad0f14ca072c?sid=77d03c3a-f8e6-46de-ad08-4749b6776175) where I explained the issue visually. It is about the comment, it adds it to only the first user, not to each user. Kindly take a double look.